tests/unit/test_network_connections.py

```diff
@@ -2792,13 +2792,171 @@ def test_route_v6(self):
                         "DEVICE": "e556",
                     },
                     "keys": None,
-                    "route": "192.168.40.0/24 metric 545\n192.168.46.0/30\n"
+                    "route": "192.168.40.0/24 dev e556 metric 545\n"
+                    "192.168.46.0/30\n"
+                    "192.168.45.0/24 dev e556 metric 545\n"
+                    "192.168.46.0/30 dev e556\n",
+                    "route6": "a:b:c:f::/64\na:b:c:d::/64 dev e556\n",
+                    "rule": None,
+                    "rule6": None,
+                }
+            ],
+        )
+
+    def test_route_v6_without_interface_name(self):
+        route_device_warning = (
+            "The connection e556 does not specify an interface name. Therefore, the "
+            "route to {} will be configured without the output device and the kernel "
+            "will choose it automatically which might result in an unwanted device "
+            "being used. To avoid this, specify `interface_name` in the connection "
+            "appropriately."
```

Comment:
Python2.6 will require you to use indices in the format specification. Please, replace `{}` with `{0}` and that should solve the unit test failing.